lib/rucio/api/did.py

```diff
@@ -66,7 +66,22 @@ def list_dids(scope, filters, did_type='collection', ignore_case=False, limit=No
 
 
 @transactional_session
-def add_did(scope, name, did_type, issuer, account=None, statuses={}, meta={}, rules=[], lifetime=None, dids=[], rse=None, vo='def', *, session: "Session"):
+def add_did(
+    scope: str,
+    name: str,
+    did_type: str,
+    issuer: str,
+    account: "Optional[str]" = None,
+    statuses: "Optional[dict[str, str]]" = None,
+    meta: "Optional[dict[str, str]]" = None,
+    rules: "Optional[list[dict[str, Any]]]" = None,
```

Comment:
`Sequence` instead of `list` to match the existing `add_did` signature in `core`: https://github.com/rucio/rucio/blob/19392c29c61021d37aa1f3bbe6017722222994cf/lib/rucio/core/did.py#L119-L131

Reply:
Changed, now: 
```
rules: "Optional[Sequence[dict[str, Any]]]" = None,
```